### ISSUE TITLE
Italic does not compile on Linux.

### ISSUE THREAD

**Issue by martinjacobd** [psb1558/Junicode-New#35]
Hi, I was able to get all widths and weights of the Roman type to compile on Debian GNU/Linux by removing the ss10 and ss20 features, which were complained about. (I am using fontmake version 2.2.0.) But I cannot get the Italic shape to compile. Here is the patch I applied to get the Roman to compile:

> diff --git a/Junicode-width.glyphs b/Junicode-width.glyphs
index ecc0094..08e7581 100644
--- a/Junicode-width.glyphs
+++ b/Junicode-width.glyphs
@@ -165,7 +165,7 @@ name = Languagesystems;
 features = (
 {
 automatic = 1;
-code = "feature locl;\012feature c2sc;\012feature smcp;\012feature pcap;\012feature case;\012feature nalt;\012feature tnum;\012feature onum;\012feature pnum;\012feature lnum;\012feature zero;\012feature sups;\012feature subs;\012feature ss01;\012feature ss02;\012feature hist;\012feature ss03;\012feature ss04;\012feature ss05;\012feature ss06;\012feature ss07;\012feature ss08;\012feature ss10;\012feature ss11;\012feature ss12;\012feature ss13;\012feature ss14;\012feature ss15;\012feature ss16;\012feature ss18;\012feature ss19;\012feature ss20;\012feature cv01;\012feature cv02;\012feature cv04;\012feature cv05;\012feature cv07;\012feature cv08;\012feature cv09;\012feature cv10;\012feature cv11;\012feature cv12;\012feature cv13;\012feature cv15;\012feature cv16;\012feature cv17;\012feature cv18;\012feature cv19;\012feature cv21;\012feature cv22;\012feature cv23;\012feature cv24;\012feature cv25;\012feature cv26;\012feature cv28;\012feature cv30;\012feature cv31;\012feature cv32;\012feature cv40;\012feature cv41;\012feature cv42;\012feature cv43;\012feature cv44;\012feature cv45;\012feature cv46;\012feature cv50;\012feature rtlm;\012feature calt;\012";
+code = "feature locl;\012feature c2sc;\012feature smcp;\012feature pcap;\012feature case;\012feature nalt;\012feature tnum;\012feature onum;\012feature pnum;\012feature lnum;\012feature zero;\012feature sups;\012feature subs;\012feature ss01;\012feature ss02;\012feature hist;\012feature ss03;\012feature ss04;\012feature ss05;\012feature ss06;\012feature ss07;\012feature ss08;\012feature ss11;\012feature ss12;\012feature ss13;\012feature ss14;\012feature ss15;\012feature ss16;\012feature ss18;\012feature ss19;\012feature cv01;\012feature cv02;\012feature cv04;\012feature cv05;\012feature cv07;\012feature cv08;\012feature cv09;\012feature cv10;\012feature cv11;\012feature cv12;\012feature cv13;\012feature cv15;\012feature cv16;\012feature cv17;\012feature cv18;\012feature cv19;\012feature cv21;\012feature cv22;\012feature cv23;\012feature cv24;\012feature cv25;\012feature cv26;\012feature cv28;\012feature cv30;\012feature cv31;\012feature cv32;\012feature cv40;\012feature cv41;\012feature cv42;\012feature cv43;\012feature cv44;\012feature cv45;\012feature cv46;\012feature cv50;\012feature rtlm;\012feature calt;\012";
 name = aalt;
 },
 {

The resulting OTFs work quite well and are very handsome!

However, the Italic fails with this error message [complete output of make after Roman has compiled]:
> fontmake -g Junicode-Italic.glyphs -i -o otf
INFO:fontmake.font_project:Building master UFOs and designspace from Glyphs source
INFO:glyphsLib.classes:Parsing "Junicode-Italic.glyphs" file into <GSFont>
WARNING:glyphsLib.builder.components:Glyph 'uni2E4E': All components of the background layer of 'Italic' will be decomposed.
WARNING:glyphsLib.builder.components:Glyph 'uni2E4E': All components of the background layer of 'Bold Italic' will be decomposed.
WARNING:glyphsLib.builder.builders.UFOBuilder:Anchors not propagated for inexistent component uniE1AC in glyph uniEBA7
WARNING:glyphsLib.builder.builders.UFOBuilder:Anchors not propagated for inexistent component oogonek in glyph uni01ED
WARNING:glyphsLib.builder.builders.UFOBuilder:Anchors not propagated for inexistent component uniE1AC in glyph uniEBA7
INFO:fontmake.font_project:Interpolating master UFOs from designspace
fontmake: Error: In 'Junicode-Italic.glyphs' -> 'master_ufo/JuniusX-Italic.designspace': Preparing the Designspace for interpolation failed: Cannot set up glyph 'g' for interpolation: Locations must be unique.'
make: *** [Makefile:3: all] Error 1

Trying to change the sensitivity of fontmake's renderers did not work, and it is impossible to use the alternate renderer mutatormath because it doesn't support all of the features JuniusX needs. 

It would be nice for this to compile on Linux. Alternatively, if we could download precompiled OTFs, that would be great! But it's awkward that we can't use this open source font without resorting to proprietary tools.

**Comment by psb1558** [comment]
With respect to the roman: your sources are out of date. For a few days in the latter half of July, there was an error in the source relating to the removal of ss10 and ss20, which I corrected a few days short of two weeks ago. To test I just compiled with
```
fontmake -o otf -i --optimize-cff 2 -g ../Junicode-width.glyphs
```
Whew! It takes patience. Generating ttf is much faster, and I recommend it if only because of the capable hinting with ttfautohint.

The italic is another story. I can't build it with either Glyphs or fontmake, and that's because it is simply not ready. As the README says, it is about 43% complete; and then there will be many details to attend to before it's Ready for Prime Time. This is why I've never done a release of JuniusX. As JuniusX/VF is a reworking of Junicode, I suggest that users go on using Junicode (making sure to pick up the latest version in the "legacy" directory of this repo). Unless you don't need an italic, in which case JuniusX is just fine (though with some rough edges).

Check back in a year to a year and a half; with any luck, the italic will be done by then.
